core/src/sqlite_workers/sqlite_database.rs

```diff
@@ -113,14 +136,23 @@ impl SqliteDatabase {
                             ))
                         })
                         .collect::<Result<serde_json::Value>>()
-                })?
-                // Limit to 128 rows.
-                .take(MAX_ROWS)
-                .collect::<Result<Vec<_>>>()?
+                })
+                // At this point we have a result (from the query_and_then fn itself) of results (for each
+                // individual row parsing). We wrap the potential top-level error in a QueryError and bubble it up.
+                .map_err(|e| QueryError::QueryExecutionError(anyhow::Error::new(e)))?
+                .take(MAX_ROWS + 1)
```

Comment:
So the hack it to keep `MAX_ROWS + 1` so we still enter the if block that triggers a `ExceededMaxRows` error, is that right?

Reply:
Yeah I take max_rows + 1 so I can have the if below. Maybe there's a way to include that into the iterator pipeline but I don't know it